R/plot.R

```diff
@@ -0,0 +1,449 @@
+
+#' A histogram of density ratio estimates
+#'
+#' Creates a histogram of the density ratio estimates. Useful to understand the
+#' distribution of estimated density ratios in each sample, or compare it among
+#' samples. It is the default plotting method for density ratio objects.
+#'
+#' @param object Density ratio object created with e.g., [kliep()], [ulsif()],
+#'   or [naive()]
+#' @param sample Character string indicating whether to plot the 'numerator',
+#'   'denominator', or 'both' samples. Default is 'both'.
+#' @param logscale Logical indicating whether to plot the density ratio
+#'   estimates on a log scale. Default is TRUE.
+#' @param binwidth Numeric indicating the width of the bins, passed on to
+#'   `ggplot2`.
+#' @param bins Numeric indicating the number of bins. Overriden by binwidth, and
+#'   passed on to `ggplot2`.
+#' @param ... Additional arguments passed on to `predict()`.
+#'
+#' @return A histogram of density ratio estimates.
+#'
+#'
+#' @examples
+dr.histogram <- function(object,
+                         samples = "both",
+                         logscale = TRUE,
+                         binwidth = NULL,
+                         bins = NULL,
+                         ...) {
+
+  # Check object type
+  check.object.type(object)
+
+
+  # Create data object and estimate density ratio
+  data <- rbind(object$df_numerator, object$df_denominator)
+  # Check no variable names that will be overriden when plotting.
+  check.overriden.names(data)
```

Comment:
Can't we create the `dr` variable outside of the data object? Thus,
```
df <- data.frame(y = rnorm(100),
                 x = rnorm(100))

ext <- list(
  dr = 1:100,
  sample = runif(100)
)

library(ggplot2)
ggplot(df, aes(x = ext$dr, y = y)) +
  geom_point()
```

if we make a separate object, we don't have to check for variable names (note that making these variables outside a list (or other type of overarching object) will not work if the variables with the same name are also in the data, because ggplot will look in the data first.

Reply:
I might be missing something about ggplot2's behaviour, but if we do as you suggest:
- We still have to check for variable names in the dataframe (because ggplot will look in the data first).
- There might be problems because the dr and the data are not linked. The correcteness of the plot relies on the vector of dr being always in the same order than the original dataframe; any reordering of the dataframe would result in the plot working but showing incorrect results (see what I mean below).  I feel like this behaviour is too bug-prone (even if as far as I can see it'd work in this case, other code changes might break it). 

So I if I understood correctly, I don't see the benefit of creating dr in a external object.

```
df <- data.frame(y = 100:1,
                 x = rnorm(100))

ext <- data.frame(
  dr = 1:100,
  sample = runif(100)
)


library(ggplot2)
ggplot(df, aes(x = ext$dr, y = y)) +
  geom_point()


df2 <- df %>% 
  dplyr::arrange(y)

ggplot(df2, aes(x = ext$dr, y = y)) +
  geom_point()
```

Reply:
All clear! ggplot2 will ofc read the correct variable, so this is an advantage (no need to change variable names).